### PR TITLE
Address the tasks in the Meta section

### PR DESCRIPTION
**Pull request by anikiki**
Task/Issue URL: https://app.asana.com/0/72649045549333/1208287265878043/f

### Description
Added feature flag for switch to tab. Removed unneeded quick insert icons. Formatted urls in autosuggest. Added dividers between autosuggest sections. 
Updated color on switch to tab icon.

### Steps to test this PR

_Feature 1_
- [x] Install from this branch.
- [x] Open a few tabs.
- [x] Search for something that matches the open tabs. Notice there's a divides between the top hits and search items. Notice there's a divider between the search items and the bottom section.
- [x] Notice the quick insert icon is available only for search suggestions.
- [x] Notice all URLs in suggestions don't have `http://www.`, `htps://www.`, `www.` as prefix and they don't have a trailing `/`.
- [x] Notice that if you turn off the `autocompleteTabs` from `Feature Flag Inventory` you don't see the Switch to tab suggestions. 

### UI changes
| Divider dark node  | Divider light mode |
| ------ | ----- |
|divider_light_mode|divider_dark_mode|

|No top divider | Bottom divider |
| ------ | ----- |
|no_top_divider|bottom_divider|

### CODE REVIEW

**Comment by anikiki** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5105?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#5105** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5105?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#5065** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5065?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5064** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5064?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5063** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5063?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5042** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5042?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5019** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5019?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5018** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5018?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @anikiki and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by CrisBarreiro — APPROVED** [review]
Added a few comments, but works as expected! 🚀

File: app/src/main/java/com/duckduckgo/app/autocomplete/api/AutoComplete.kt [hunk 1]
```diff
@@ -247,13 +249,18 @@ class AutoCompleteApi @Inject constructor(
...
     private fun getAutocompleteSwitchToTabResults(query: String): Observable<MutableList<RankedSuggestion<AutoCompleteSwitchToTabSuggestion>>> =
-        tabRepository.getTabsObservable()
-            .map { rankTabs(query, it) }
-            .flattenAsObservable { it }
-            .distinctUntilChanged()
-            .toList()
-            .onErrorReturn { emptyList() }
-            .toObservable()
+        // TODO: ANA - Do we want to have this check here, or somewhere else? (note: this is using the RxComputationThreadPool).
+        if (autocompleteTabsFeature.self().isEnabled()) {
```
[the remark this answers is not in the document]
Comment (anikiki) [reply app/src/main/java/com/duckduckgo/app/autocomplete/api/AutoComplete.kt:253]: I updated this in the last PR -> https://github.com/duckduckgo/Android/pull/5123/files#diff-e563a062fe06ca0a38b0c058df8314a24f2fffb40952a85a1e356be04697b58bR264

File: app/src/main/java/com/duckduckgo/app/autocomplete/api/AutoComplete.kt [hunk 2]
```diff
@@ -247,13 +249,18 @@ class AutoCompleteApi @Inject constructor(
     }
 
     private fun getAutocompleteSwitchToTabResults(query: String): Observable<MutableList<RankedSuggestion<AutoCompleteSwitchToTabSuggestion>>> =
```
[the remark this answers is not in the document]
Comment (anikiki) [reply app/src/main/java/com/duckduckgo/app/autocomplete/api/AutoComplete.kt:251]: All `MutableList` were removed after moving away from rxJava -> https://github.com/duckduckgo/Android/pull/5123

File: app/src/test/java/com/duckduckgo/app/autocomplete/api/AutoCompleteApiTest.kt
[the remark this answers is not in the document]
Comment (anikiki) [reply app/src/test/java/com/duckduckgo/app/autocomplete/api/AutoCompleteApiTest.kt:1]: Added :+1:

File: app/src/main/java/com/duckduckgo/app/browser/autocomplete/SuggestionViewHolderFactory.kt [hunk 3]
```diff
@@ -262,7 +260,7 @@ sealed class AutoCompleteViewHolder(itemView: View) : RecyclerView.ViewHolder(it
             editableSearchClickListener: (AutoCompleteSuggestion) -> Unit,
             omnibarPosition: OmnibarPosition,
         ) = with(binding) {
-            phrase.text = item.phrase
+            phrase.text = item.phrase.formatIfUrl()
```
[the remark this answers is not in the document]
Comment (anikiki) [reply app/src/main/java/com/duckduckgo/app/browser/autocomplete/SuggestionViewHolderFactory.kt:263]: Done in the final PR -> https://github.com/duckduckgo/Android/pull/5123

**Pull request merged**